storage_mem.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"fmt"
+
+	model "github.com/golang-bristol/beer-model"
+)
+
+// StorageMemory data storage type save only in memory
+type StorageMemory struct {
+	cellar  []model.Beer
+	reviews []model.Review
+}
+
+func newStorageMemory() (*StorageMemory, error) {
+	var err error
+
+	stg := new(StorageMemory)
+
+	return stg, err
+}
+
+// SaveBeer insert or update beers
+func (s *StorageMemory) SaveBeer(beers ...model.Beer) error {
+	for _, beer := range beers {
+		var err error
+
+		beersFound, err := s.FindBeer(model.Beer{ID: beer.ID})
+		if err != nil {
+			return err
+		}
+
+		if len(beersFound) == 1 {
+			*beersFound[0] = beer
+
+		} else if len(beersFound) > 1 {
+			return fmt.Errorf("can't update because found multiple entries with ID %d", beer.ID)
+		}
+
+		s.cellar = append(s.cellar, beer)
+	}
+
+	return nil
+}
+
+// SaveReview insert or update reviews
+func (s *StorageMemory) SaveReview(reviews ...model.Review) error {
+	for _, review := range reviews {
+		var err error
+
+		reviewsFound, err := s.FindReview(model.Review{ID: review.ID})
+		if err != nil {
+			return err
+		}
+
+		if len(reviewsFound) == 1 {
+			*reviewsFound[0] = review
+
+		} else if len(reviewsFound) > 1 {
+			return fmt.Errorf("can't update because found multiple entries with ID %d", review.ID)
+		}
+
+		s.reviews = append(s.reviews, review)
+	}
+
+	return nil
+}
+
+// FindBeer locate full data set based on given criteria
+func (s *StorageMemory) FindBeer(criteria model.Beer) ([]*model.Beer, error) {
```

Comment:
I think it will be more efficient to have separate functions for FindBeers and FindBeer. The reason being a) not having to use the blank beer as a contract for "return all" and b) FindBeers will be much more efficient if you return all data right away rather than evaluating the `if criteria == blankBeer {` every time for each beer.